inductiva/tasks/task.py

```diff
@@ -0,0 +1,147 @@
+"""Interact with a running/completed task on the Inductiva API."""
+import pathlib
+import time
+from typing import Dict, Any
+from typing_extensions import TypedDict
+
+from inductiva.client.models import TaskStatusCode
+from inductiva import api
+from inductiva.client.apis.tags.tasks_api import TasksApi
+from absl import logging
+
+
+class Task:
+    """Represents a running/completed task on the Inductiva API.
+
+    Attributes:
+        id: The task ID.
+        _api: Instance of TasksApi (from the generated API client).
+    """
+
+    def __init__(self, task_id: str):
+        """Initialize the instance from a task ID."""
+        self.id = task_id
+        self._api = TasksApi(api.get_client())
+
+    def __enter__(self):
+        """Enter context manager for managing a blocking execution.
+
+        If an exception/ctrl+c is caught while in the context manager, the task
+        is killed.
+
+        Usage:
+            task = scenario.simulate_async(...)
+            with task:
+                # If an exception happens here or ctrl+c is pressed, the task
+                # will be killed.
+                task.wait()
+
+        """
+        return self
+
+    def __exit__(self, exc_type, exc_value, traceback):
+        """Exit context manager killing the task if an exception was raised."""
+        del traceback  # unused
+        del exc_type  # unused
+
+        if exc_value is None:
+            return True
+
+        if isinstance(exc_value, KeyboardInterrupt):
+            logging.info("Caught SIGINT: terminating blocking task...")
+        elif exc_value is not None:
+            logging.info("Caught exception: terminating blocking task...")
+
+        self.kill()
+        return False
+
+    def get_status(self) -> TaskStatusCode:
+        """Get status of the task.
+
+        This method issues a request to the API.
+        """
+        resp = self._api.get_task_status(self._get_path_params())
```

Comment:
This is querying the redis stream right? Should it query the database? What is the correct option?

Reply:
This is sending a request to the API, without any awareness of what the API uses underneath to serve the request.